src/chai/ManagedArray.inl

```diff
@@ -613,7 +613,7 @@ template<typename T>
 CHAI_INLINE
 CHAI_HOST_DEVICE
 bool
-ManagedArray<T>::operator== (ManagedArray<T>& rhs) const
+ManagedArray<T>::operator== (const ManagedArray<T>& rhs) const
 {
   return (m_active_pointer ==  rhs.m_active_pointer);
```

Comment:
I'm also wondering if this should be `m_pointer_record == rhs.m_pointer_record`. Or if the host case should pull the host pointer from the pointer record: `m_pointer_record->m_pointers[CPU] == rhs.m_pointer_record->m_pointers[CPU]`